### ISSUE TITLE
[Propuesta de actividad] Taller de Introducción a JHipster

### ISSUE THREAD

**Issue by girianshido** [gpul-labs/roadmap#43]
Taller de introducción al generador de Yeoman [JHipster](http://jhipster.github.io/) para crear un proyecto web con una API RESTful de lado servidor (Spring) y una Single Web page del lado cliente (AngularJs).

**Comment by ssaavedra** [comment]
Pensa que para que esto forme parte do proxecto debería de integrarse dalgún xeito co resto do mesmo (é dicir, axudar a que constrúa algún "módulo" ou funcionalidade que antes non teñamos), e que, moi probablemente, implique que o código do lado servidor se execute na Raspberry.
Queremos executar Spring nunha RPi? :disappointed_relieved:

**Comment by castrinho8** [comment]
A min isto paréceme interesante como extra, non pensando no proxecto se non en unha alternativa para crear webs a que se propón no fío xeral do proxecto con Django ;)
Igual podemos facer unha semana dos frameworks web e ter dúas charlas esa semana jajaja

**Comment by girianshido** [comment]
@castrinho8 Sí, a proposta sería para un tema adicional aos Labs. Estaría xenial esa semana adicional de charlas para diferentes Frameworks. Jhipster en realidade é un xerador de [Yeoman](http://yeoman.io/) que integra múltiples tecnoloxías para crear webs modernas e está todos os días actualizándose e engadíndose novas funcionalidades.

@ssaavedra a RPi pode levantar un proxecto empaquetado en war de JHipster co SDK de Java8, tamén incorpora un modo de despliegue co servidor lixeiro [Undertow](http://undertow.io/). Claro que a RPi non creo que soportase moita carga de usuarios e peticións. 

Como idea poderíamos facer probas de testing e rendemento con Apache JMeter ou incluso chegar a probar cun clúster de [48 RPi para Tomcat](https://erikwramner.files.wordpress.com/2013/09/con4177_wramner.pdf).

**Comment by ssaavedra** [comment]
> chegar a probar cun clúster de 48 RPi para Tomcat.

Citando ao propio autor, eso ten un coste (eu diría como mínimo) duns $3000, que se nos vai de orzamento para unha actividade :-)

Non dubido que a raspi non sexa capaz de correr Java aceptablemente, pero seguro que se o fai non pode facer nada máis (como ter ademais o Postgres, o server wsgi de Django, un nginx de front...) :-)

Por outra banda, refíreste a que habería que usar a JVM de Oracle (en lugar da OpenJDK)? Porque iso non o pode promulgar GPUL. Temos un compromiso co software libre; e de feito se algo é razoable debería de funcionar na OpenJDK, xa que é a implementación de referencia [[[1](https://blogs.oracle.com/henrik/entry/moving_to_openjdk_as_the)]](https://blogs.oracle.com/henrik/entry/moving_to_openjdk_as_the).

Pero como dicides, eu esto vexo que encaixa máis fóra, como unha cousa independente que poida executarse en procesadores máis _featureful_.
